### ISSUE TITLE
Improving developer on-boarding experience with a CLI command

### ISSUE THREAD

**Issue by taras** [resideo-platform/exemplar#52]
We want it to be easy for a new developer to join the project. They should be able to get set up, run tests and be able to create new PR in less than an hour. Today, the chances of this happening are low because lack of documentation and the setup process is not well automated. 

We need an automated onboarding experience that will automatically configure the development environment including everything that a developer needs to be productive on this project.

It is inevitable that some steps of the build process will fail. The tooling provided for developers should help them identify problems, suggest solutions and verify resolution.

The setup process should be broken up into steps. Each step should be executable automatically and manually when required. The steps should clearly articulate how they verify a successful execution of the step and how it accomplishes this step. 

The setup mechanism should be able to read the environment and adjust its process to match. For example, if the developer is using `bash` or `zsh`, their environment variables would be written to different files. 

I propose that we create a CLI tool with a `setup` command that will take arguments that will perform the setup for the project. This tool will supersede tools like `nativescript-cli` because it'll include recommendations for our particular project rather than the general recommendation for NativeScript.

For this particular project, the setup process would start by executing all of the steps described in the [Advanced setup of NativeScript CLI](https://docs.nativescript.org/start/ns-setup-os-x). It should allow configuration for the development environment for iOS development without installing dependencies for Android and vice-versa. The ability to configure for iOS or Android without requiring both will give developers an opportunity to opt-in to setting up the other based on their circumstances.

**Comment by cowboyd** [comment]
What if we started by focusing first and foremost on _diagnosing_ exactly why something isn't working. And only then, perhaps trying to automatically fix it, but even then we have to be very careful about what we choose to automate. The reason is because the minute to start to do configuration management is that you're now assuming the role of system configuration management tools like `puppet`, `salt`, `chef`, and `docker`, `apt`, `npm`. And the reason all those tools are complex is that the problem space is fractally complex.

A great example of how _not_ to do it is the NativeScript CLI which, although a fantastic project, with really smart developers, kinda [missed the boat][1] with their attempts to set up systems automatically using the `tns doctor` command.

I used this at the beginning of this project, and it screwed up my system and ended up costing me much more time than it saved.

What if we split it this into two tasks. Diagnosis and remedy, and did the diagnosis first.

[1]: https://github.com/NativeScript/nativescript-cli/issues/4382

**Comment by taras** [comment]
I agree that these steps need to be separate. Let's start by defining what the CLI API would be and then we can design the actual implementation. I'm thinking that it would look something like this,

```bash
resideo check ios
```

It would generate a list of things that the system needs to have iOS working. 

```js
[ ] Homebrew installed
[ ] Node > 10
[ ] Latest version of XCode Installed
[ ] Command-line tools for Xcode
[ ] xcodeproj ruby gem
[ ] CocoaPods
[ ] The six python package
```

What would be the next step in this process?

**Comment by cowboyd** [comment]
I'd say the first thing would be to leverage the Nativescript checks, and then start implementing our own.

I _think_ that the if you run `NS_SKIP_ENV_CHECK=true yarn tns doctor` it will do the checks, it will only print warnings.
